cmd/kured/main.go

```diff
@@ -493,6 +553,12 @@ func rebootAsRequired(nodeID string, rebootCommand []string, sentinelCommand []s
 		log.Fatal(err)
 	}
 
+	if labelWithELBX {
```

Comment:
Don't we want to move this remove node label logic down into the the part where we know we're holding the lock? Similar to where we do `if annotateNodes && !rebootRequired` and then delete node annotations?

Reply:
The `disableExcludeFromELBs()` method can be run safely with or without the lock. Are there any particular cases you're thinking of where you would want the method to run only if the lock is present?

Reply:
I'm just thinking that if we do this when we're not holding the lock, that means we're doing it in every iteration. And that means that we're calling the apiserver for the node object every time. On a cluster with hundreds of thousands of nodes, with a small kured interval (e.g., 1 min), that means we're going to potentially be sending many apiserver requests every second.

If I understand correctly, the only reason we want to disable this label is to remove it after a successful reboot. In every scenario that fits that description, we *should* have the lock (because we only ever release the lock inside that block).

Does that make sense?

cc @dholbach @ckotzbauer for thoughts as well

Reply:
I agree with @jackfrancis, that we should avoid requesting the apiserver to excessive. The logic should be safe when only called with a lock being present.